cnf-tests/README.md

```diff
@@ -0,0 +1,134 @@
+kind: MachineConfigPool
+metadata:
+  name: worker-cnf
+  labels:
+    machineconfiguration.openshift.io/role: worker-cnf
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {
+          key: machineconfiguration.openshift.io/role,
+          operator: In,
+          values: [worker-cnf, worker],
+        }
+  paused: false
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker-cnf: ""
+```
+
+The worker pool name can be overridden via the `ROLE_WORKER_CNF` variable.
+
+```bash
+    docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig -e ROLE_WORKER_CNF=custom-worker-pool quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh
+```
+
+Please note that currently not all tests run selectively on the nodes belonging to the pool.
+
+## Gingko Parameters
+
+The test suite is built upon the ginkgo bdd framework.
+This means that it accept parameters for filtering or skipping tests.
+
+To filter a set of tests, the -ginkgo.focus parameter can be added:
+
+```bash
+    docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh -ginkgo.focus="performance|sctp"
+```
+
+### Available features
+
+The set of available features to filter are:
+
+- performance
+- sriov
+- ptp
+- sctp
+- dpdk
+
+### Dry Run
+
+To run in dry-run mode:
+
+```bash
+    docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh -ginkgo.dryRun -ginkgo.v
+```
+
+## Disconnected Mode
+
+The CNF tests image support running tests in a disconnected cluster, meaning a cluster that is not able to reach outer registries.
+This is done in two steps:
+
+### Mirroring the images to a custom registry accessible from the cluster
+
+A `mirror` executable is shipped in the image to provide the input required by oc to mirror the images needed to run the tests to a local registry.
+
+The following command
+
+```bash
+    docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig quay.io/openshift-kni/cnf-tests /usr/bin/mirror -registry my.local.registry:5000/ |  oc image mirror -f -
+```
+
+can be run from an intermediate machine that has access both to the cluster and to quay.io over the Internet.
+
+### Instruct the tests to consume those images from a custom registry
+
+This is done by setting the IMAGE_REGISTRY environment variable:
+
+```bash
+    docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig -e IMAGE_REGISTRY my.local.registry:5000/ quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh
+```
+
+## Test Reports
+
+The tests have two kind of outputs
+
+### JUnit test output
+
+A junit compliant xml is produced by passing the `-juni`t parameter together with the path where the report is dumped:
+
+```bash
+    docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh -junit /path/to/junit
```

Comment:
I think you need to add here another volume to take the junit file from inside the container